src/main/scala/com/advancedtelematic/director/db/DeviceUpdate.scala

```diff
@@ -69,21 +70,30 @@ object DeviceUpdate extends AdminRepositorySupport
     db.run(dbAct.transactionally)
   }
 
-  private [db] def clearTargetsFromAction(namespace: Namespace, device: DeviceId, version: Int)
+  private def copyTargetsAction(namespace: Namespace, device: DeviceId, failedTargets: Map[EcuSerial, TargetFilename],
+                                latestVersion: Int, nextTimestampVersion: Int)(implicit db: Database, ec: ExecutionContext) =
+    if (failedTargets.isEmpty) {
+      adminRepository.copyTargetsAction(namespace, device, latestVersion, nextTimestampVersion)
+    } else {
+      adminRepository.copyTargetsAction(namespace, device, nextTimestampVersion - 1, nextTimestampVersion, failedTargets)
```

Comment:
isn't this latestVersion ?

Reply:
No, it's a _different_ one in this method. I"ll rename them to make it clearer.